src/translations/mapCollection.js

```diff
@@ -0,0 +1,122 @@
+import { isUndefined, isDefined } from '../helpers/mixed';
+import { mixin } from '../helpers/object';
+import localHooks from '../mixins/localHooks';
+
+// Counter for checking if there is a memory leak.
+let registeredMaps = 0;
+
+/**
+ * Collection of maps.
+ */
+class MapCollection {
+  constructor() {
+    /**
+     * Mappings from the unique name to the stored map.
+     *
+     * @type {Map<string, BaseMap>}
+     */
+    this.mappings = new Map();
+  }
+
+  /**
+   * Register custom index map.
+   *
+   * @param {String} uniqueName Unique name of the map.
+   * @param {BaseMap} map Map containing miscellaneous (i.e. meta data, indexes sequence), updated after remove and insert data actions.
+   * @returns {BaseMap|undefined}
+   */
+  register(uniqueName, map) {
+    if (this.mappings.has(uniqueName) === false) {
+      this.mappings.set(uniqueName, map);
+
+      map.addLocalHook('change', () => this.runLocalHooks('change', map));
+
+      registeredMaps += 1;
+    }
+  }
+
+  /**
+   * Unregister custom index map.
+   *
+   * @param {String} name Name of the map.
+   */
+  unregister(name) {
+    const map = this.mappings.get(name);
+
+    if (isDefined(map)) {
+      map.clearLocalHooks();
+      this.mappings.delete(name);
+
+      this.runLocalHooks('change', map);
+
+      registeredMaps -= 1;
+    }
+  }
+
+  /**
+   * Get indexes list for provided index map name.
```

Comment:
I think this method should be broken into two methods:

```
getMap(name)
getMaps()
```

A function should do one thing only. That is good for programmer's understanding, as well as for runtime performance.